src/Symfony/Component/Serializer/Normalizer/BackedEnumNormalizer.php

```diff
@@ -63,7 +63,7 @@ public function denormalize($data, string $type, string $format = null, array $c
 
         try {
             return $type::from($data);
-        } catch (\ValueError $e) {
+        } catch (\ValueError|\TypeError $e) {
```

Comment:
To me, the TypeError should result in a NotNormalizableValueException, as it is a more precise case of the check above (as an integer-backed enum won't support strings but the normalizer does not distinguish the backing type)

Reply:
@stof Now, TypeError result in a NotNormalizableValueException.